src/target/riscv/riscv-013.c

```diff
@@ -2924,22 +2924,50 @@ static int deassert_reset(struct target *target)
 	riscv_scan_set_delay(&info->learned_delays, RISCV_DELAY_BASE,
 			orig_base_delay);
 
-	/* Ack reset and clear DM_DMCONTROL_HALTREQ if previously set */
+	/* Ack reset with DM_DMCONTROL_HALTREQ for those MCUs which need a
```

Comment:
I believe that the original comment should stay as is, and the phrase about 

```
period of time to be halted after reset is released 
```
should be moved to the wait loop.

Reply:
I'm okay to move "wait for a period of time" into while, but to keep "clear DM_DMCONTROL_HALTREQ" as is perhaps not aligned with current code, as that bit will be cleared later. how about change the line of 2927 as following:
`/* Act reset with DM_DMCONTROL_HALTREQ based on target->reset_haltreq */`

Reply:
Thank you, @JanMatCodasip. Agree, comments should be clear enough.